src/backend/base/langflow/components/models/AIMLModel.py

```diff
@@ -71,24 +64,31 @@ class AIMLModelComponent(LCModelComponent):
         ),
     ]
 
+    def update_build_config(self, build_config: dict, field_name: str | None = None):
+        if field_name == "api_key" or field_name == "aiml_api_base" or field_name == "model_name":
+            aiml = AimlModels()
+            aiml.get_aiml_models()
+            build_config["model_name"]["options"] = aiml.chat_models
+        return build_config
+
     def build_model(self) -> LanguageModel:  # type: ignore[type-var]
         aiml_api_key = self.api_key
         temperature = self.temperature
         model_name: str = self.model_name
         max_tokens = self.max_tokens
         model_kwargs = self.model_kwargs or {}
-        aiml_api_base = self.aiml_api_base or "https://api.aimlapi.com"
-        seed = self.seed
+        aiml_api_base = self.aiml_api_base or "https://api-staging.aimlapi.com/v2"
 
         openai_api_key = aiml_api_key.get_secret_value() if isinstance(aiml_api_key, SecretStr) else aiml_api_key
 
+        if "o1" in model_name:
+            temperature = 1
```

Comment:
Why do this?

Reply:
Since models starting with "o1" don’t work with temperatures other than 1, we need to hard-code this value for these models. I believe the best approach is to fix our API, and I’ve already asked my coworkers if we can proceed with this solution.

Reply:
Makes sense - do you mind adding a comment here to remind me in the future?